src/components/Settings/SettingsImportContacts.vue

```diff
@@ -23,21 +23,87 @@
-			</label>
+			<button @click="toggleModal">
+				{{ t('contacts', 'Import contacts') }}
+			</button>
+			<!-- <div>
+				<input id="contact-import"
+					:disabled="isImporting"
+					type="file"
+					class="hidden-visually"
+					@change="processFile">
+				<label id="upload" for="contact-import" class="button import-contact__multiselect-label icon-upload">
+					{{ t('contacts', 'Select local file') }}
+				</label>
+				<button id="pick" class="button import-contact__multiselect-label icon-folder">
+					{{ t('contacts', 'Select from files') }}
+				</button>
+			</div>
 			<multiselect
 				v-model="selectedAddressbook"
 				:options="options"
 				:disabled="isSingleAddressbook || isImporting"
 				:placeholder="t('contacts', 'Contacts')"
 				label="displayName"
-				class="import-contact__multiselect" />
+				class="import-contact__multiselect">
+				<template slot="singleLabel" slot-scope="{option}">
+					{{
+						isImporting
+							? t('contacts', `Importing into '{addressbook}'`, { addressbook: option.displayName })
+							: t('contacts', `Import into '{addressbook}'`, { addressbook: option.displayName })
+					}}
+				</template>
+			</multiselect> -->
+			<Modal v-if="isOpened"
+				ref="modal"
+				class="import-contact__modal"
+				:title="t('contacts', 'Import contacts')"
+				@close="toggleModal">
+				<section class="import-contact__modal-addressbook">
+					<h3>
+						{{ t('contacts', `Importing contacts into the '{addressbookName}' addressbook`, { addressbookName: selectedAddressbook.displayName }) }}
+					</h3>
+					<multiselect
+						v-if="!isSingleAddressbook"
+						v-model="selectedAddressbook"
+						:options="options"
+						:disabled="isSingleAddressbook || isImporting"
+						:placeholder="t('contacts', 'Contacts')"
+						label="displayName"
+						class="import-contact__multiselect">
+						<template slot="singleLabel">
+							{{ t('contacts', 'Change default addressbook') }}
+						</template>
+					</multiselect>
+				</section>
+				<section class="import-contact__modal-pick">
+					<input id="contact-import"
+						ref="contact-import-input"
+						:disabled="loading || isImporting"
+						type="file"
+						class="hidden-visually"
+						@change="processFile">
+					<button
+						:disabled="loading"
+						class="button import-contact__button import-contact__button--local"
+						@click="clickImportInput">
+						<span class="import-contact__button-icon icon-upload" />
+						{{ t('contacts', 'Select local file') }}
+					</button>
+					<button
+						:class="{'icon-loading': loading}"
+						:disabled="loading"
+						class="button import-contact__button import-contact__button--files"
+						@click="openPicker">
+						<span class="import-contact__button-icon icon-folder" />
+						{{ t('contacts', 'Import from files') }}
+					</button>
+					<!-- <button id="cancel" class="button import-contact__button import-contact__button--cancel" @click="toggleModal">
```

Comment:
Not sure about this one, Jan, any thoughts?
Maybe not as big and a simple `Cancel` link under the buttons on the bottom right would be enough?